url354ce1a2d1_dagaren_gladiator_chess_tgz-pJ8-es_dagaren_gladiator_main_MainJ8/src/es/dagaren/gladiator/representation/BitboardPosition.java

```diff
@@ -0,0 +1,2907 @@
+               bbAdvances = bbBlackPawnReach[pieceSquareIndex];
+               bbAdvances = bbAdvances & ~(bbOccupation | ((bbOccupation ^ bbSquare[pieceSquareIndex]) >>8));
+            }
+            
+            bbAttacked |= bbAdvances;
+            
+            bbEnPassantCapture = 0;
+            //Se comprueba si es posible la captura al paso
+            if(enPassantSquare != null)
+            {
+               bbAttackedBefore = bbAttacked;
+               bbAttacked |=    bbAttackedFromSquare[pieceSquareIndex] &
+                                bbSquare[enPassantSquare.index];
+               if(bbAttackedBefore != bbAttacked)
+               {
+                  bbEnPassantCapture = bbSquare[enPassantSquareSource[enPassantSquare.index].index];
+               }
+            }
+            //////////////////////////////
+            
+            //Se comprueba si hay conexión entre la casilla del rey y la casilla de la pieza
+            //Si no la hay se generan los movimientos de forma normal
+            bbConnection = bbInclusiveConnection[pieceSquareIndex][kingSquareIndex];
+            if(bbConnection != 0)
+            {
+               //Si hay conexión, se comprueba si hay alguna pieza en las casillas
+               //que unen la casilla del rey y la casilla de la pieza
+               //Si sí que hay alguna pieza se generan los movimientos de la pieza de forma normal
+               bbConnection = bbExclusiveConnection[pieceSquareIndex][kingSquareIndex];
+               if((bbConnection & bbOccupation) == 0)
+               {
+                  //Si no hay casillas se comprueba si hay alguna pieza de ataque de largo alcance
+                  //que ataque a la pieza en la misma línea que el rey y la pieza
+                  //Si no hay ninguna se generan los movimientos de la pieza de forma norma
+                  bbAttacker = bbFullConnection[pieceSquareIndex][kingSquareIndex] & 
+                        bbAttackerToSquare[pieceSquareIndex] & bbOppositeOccupied;
+                  if(bbAttacker != 0)
+                  {
+                     genericPiece = pieceInSquare[Long.numberOfTrailingZeros(bbAttacker)].genericPiece;
+                     if(genericPiece != GenericPiece.PAWN && genericPiece != GenericPiece.KING)
+                     {
+                        //Si hay alguna pieza atacando en esa línea quiere decir que la pieza
+                        //sólo se podrá mover a las casillas entre la pieza clavada y el rey y las 
+                        //que se encuentran entre la pieza clavada y la pieza que provoca la clavada
+                        bbConnection = bbExclusiveConnection[pieceSquareIndex][Long.numberOfTrailingZeros(bbAttacker)] | bbAttacker;
+                        bbConnection |= bbExclusiveConnection[pieceSquareIndex][kingSquareIndex];
+                        
+                        bbAttacked = bbAttacked & bbConnection;
+                     }
+                  }
+               }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 2457:2478, 2586:2607.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.